src/Plutarch/Extra/Ord.hs

```diff
@@ -0,0 +1,280 @@
+        PLT -> pcon PLT
+        PEQ -> y
+        PGT -> pcon PGT
+    stimes = stimesIdempotentMonoid
+
+-- | @since 3.4.0
+instance Monoid (Term s POrdering) where
+    mempty = pcon PEQ
+
+-- TODO: PShow, PPartialOrd, POrd
+
+-- | @since 3.4.0
+newtype PComparator (a :: S -> Type) (s :: S)
+    = PComparator (Term s (a :--> a :--> POrdering))
+    deriving stock
+        ( -- | @since 3.4.0
+          Generic
+        )
+    deriving anyclass
+        ( -- | @since 3.4.0
+          PlutusType
+        )
+
+-- | @since 3.4.0
+instance DerivePlutusType (PComparator a) where
+    type DPTStrat _ = PlutusTypeNewtype
+
+{- | Given a type with a 'POrd' instance, construct a 'PComparator' from that
+ instance.
+
+ @since 3.4.0
+-}
+pfromOrd ::
+    forall (a :: S -> Type) (s :: S).
+    (POrd a) =>
+    Term s (PComparator a)
+pfromOrd = pcon . PComparator $
+    phoistAcyclic $
+        plam $ \x y ->
+            pif (x #== y) (pcon PEQ) (pif (x #< y) (pcon PLT) (pcon PGT))
+
+{- | As 'pfromOrd', but instead uses a projection function into the 'POrd'
+ instance to construct the 'PComparator'. Allows other \'-by\' behaviours.
+
+ @since 3.4.0
+-}
+pfromOrdBy ::
+    forall (a :: S -> Type) (b :: S -> Type) (s :: S).
+    (POrd a) =>
+    Term s ((b :--> a) :--> PComparator b)
+pfromOrdBy = phoistAcyclic $
+    plam $ \f -> pcon . PComparator $
+        plam $ \x y ->
+            plet (f # x) $ \fx ->
+                plet (f # y) $ \fy ->
+                    pif (fx #== fy) (pcon PEQ) (pif (fx #< fy) (pcon PLT) (pcon PGT))
+
+{- | Given a way of \'separating\' a @c@ into an @a@ and a @b@, as well as
+ 'PComparator's for @a@ and @b@, make a 'PComparator' for @c@.
+
+ = Note
+
+ This uses the fact that 'POrdering' is a 'Semigroup', and assumes that @c@ is
+ a tuple of @a@ and @b@ in some sense, and that it should be ordered
+ lexicographically on that basis.
+
+ @since 3.4.0
+-}
+pproductComparator ::
+    forall (a :: S -> Type) (b :: S -> Type) (c :: S -> Type) (s :: S).
+    Term s ((c :--> PPair a b) :--> PComparator a :--> PComparator b :--> PComparator c)
+pproductComparator = phoistAcyclic $
+    plam $ \split cmpA cmpB ->
+        pmatch cmpA $ \(PComparator fA) ->
+            pmatch cmpB $ \(PComparator fB) ->
+                pcon . PComparator . plam $ \x y ->
+                    pmatch (split # x) $ \(PPair xA xB) ->
+                        pmatch (split # y) $ \(PPair yA yB) ->
+                            (fA # xA # yA) <> (fB # xB # yB)
+
+{- | Given a way of \'discriminating\' a @c@ into either an @a@ or a @b@, as
+ well as 'PComparator's for @a@ and @b@, make a 'PComparator' for @c@.
+
+ = Note
+
+ This assumes that \'@c@s that are @a@s\' should be ordered before \'@c@s that
+ are @b@s\'.
+
+ @since 3.4.0
+-}
+psumComparator ::
+    forall (a :: S -> Type) (b :: S -> Type) (c :: S -> Type) (s :: S).
+    Term s ((c :--> PEither a b) :--> PComparator a :--> PComparator b :--> PComparator c)
+psumComparator = phoistAcyclic $
```

Comment:
`TermCont` might make a bit more cleaner code. But it's good as is.

Reply:
Actually, since I see a lot of `pmatch comp $ \(PComparator x) -> ...` pattern, it might be a good idea to make a function for this. 

Or, `pto` works as well.

Reply:
`pto` is probably what I was looking for, thanks!

Reply:
`pto` is so helpful but I always forget it's there when I need it...